File: app-miyou/src/main/java/com/miyou/app/domain/dialogue/port/PromptTemplatePort.java
package com.miyou.app.domain.dialogue.port;

import com.miyou.app.domain.dialogue.model.ConversationContext;
import com.miyou.app.domain.retrieval.model.RetrievalContext;

public interface PromptTemplatePort {
	String buildPrompt(RetrievalContext context);

	String buildPromptWithConversation(RetrievalContext context,
		ConversationContext conversationContext);

	String buildDefaultPrompt();
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/dialogue/port/PromptTemplatePort.kt
package com.miyou.app.domain.dialogue.port

import com.miyou.app.domain.dialogue.model.ConversationContext
import com.miyou.app.domain.dialogue.model.PromptRetrievalInput

interface PromptTemplatePort {
    fun buildPrompt(input: PromptRetrievalInput): String

    fun buildPromptWithConversation(
        input: PromptRetrievalInput,
        conversationContext: ConversationContext,
    ): String

    fun buildDefaultPrompt(): String
}
